restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1727,11 +1789,39 @@ private void outboundToPstn(final CreateCall request, final ActorRef sender) thr
 
 
     private void outboundToSip(final CreateCall request, final ActorRef sender) throws ServletParseException {
+        final String uri = (request.getOutboundProxy()!= null && (!request.getOutboundProxy().isEmpty())) ? request.getOutboundProxy() : "";
         SipURI outboundIntf = null;
         SipURI from = null;
         SipURI to = null;
 
-        to = (SipURI) sipFactory.createURI(request.to());
+        String toUser = "";
+        String toHost = "";
+        String scheme = "";
+        boolean isSecure = false;
+        if(request.to().contains("@") &&
```

Comment:
@abdulazizali77 You can do all using the SipURI, check javadoc here https://docs.oracle.com/cd/E28280_01/apirefs.1111/e17883/javax/servlet/sip/SipURI.html

The `to = (SipURI) sipFactory.createURI(request.to());` will check that we have `sips` and will make the new SipURI secure.
Also, you can change the host part after you create the SipURI using `to.setHost()`.